### PR TITLE
added script to fetch the data from silver to drishya_gold_table

### PR DESCRIPTION
**Pull request by drishyatm**
This script will fetch the card details for the timeframe ( start date and end date ) with respect to the user as member in the card . Gold card will have the board id , name of the card, Last updated date and board name will be saved to gold tables .

Added file:
drishya_gold_fetch_data.py

How do we test it:
Run from branch 
python delta-lake-trello/jobs/drishya_fetch_gold_cards.py --start_date 2022-01-01 --end_date 2023-02-24 --name Drishya

Selection_095

### CODE REVIEW

File: jobs/drishya_gold_fetch_data.py [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+"""
+
+import os
+import sys
+from datetime import datetime, timedelta
+import argparse
+from loguru import logger
+from pyspark.sql.functions import col
+sys.path.append(os.path.dirname(os.path.dirname(os.path.abspath(__file__))))
+from helpers import spark_helper as sh, common_functions as cf,trello_functions as tf
```
Comment (sravantit25) [inline jobs/drishya_gold_fetch_data.py]: remove common_functions and trello_functions, they are not needed here.

File: jobs/drishya_gold_fetch_data.py [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+from config import delta_table_conf as dc
+
+
+def get_user_activity_cards(spark, start_date, end_date, nameofuser):
```
Comment (sravantit25) [inline jobs/drishya_gold_fetch_data.py]: please default nameofuser to your name.

File: jobs/drishya_gold_fetch_data.py [hunk 3]
```diff
@@ -0,0 +1,92 @@
+"""
```
Comment (sravantit25) [inline jobs/drishya_gold_fetch_data.py]: Can you name the script as drishya_cards_gold.py

File: jobs/drishya_gold_fetch_data.py [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+
+        # Convert dates to a datetime object
+        start_date = datetime.strptime(start_date, '%Y-%m-%d')
+        end_date = datetime.strptime(end_date, '%Y-%m-%d')
+
+        # Fetch all cards that are for the user within the specified dates 
+        user_cards = refined_cards_data.filter((col("card_members").contains(nameofuser)) & (col("dateLastActivity") >= start_date) & (col("dateLastActivity") <= end_date))
+
+        #Select the required columns
+        user_only_cards = user_cards.select(
```
Comment (sravantit25) [inline jobs/drishya_gold_fetch_data.py]: can you fetch card_members as well here.

File: jobs/drishya_gold_fetch_data.py [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+            f'Failure in job to fetch {name_user} cards data for board')
+        raise error
+
+
+# --------START OF SCRIPT
+if __name__ == "__main__":
+    PARSER = argparse.ArgumentParser(description='Create gold delta table for User')
+    PARSER.add_argument('--start_date',type=str,required=True,help="Start Date format YYYY-MM-DD")
+    PARSER.add_argument('--end_date',type=str,required=True,help="Start Date format YYYY-MM-DD")
+    PARSER.add_argument('--name',type=str,required=True,help="Name to Fetch the Gold cards")
```
Comment (sravantit25) [inline jobs/drishya_gold_fetch_data.py]: Please remove this argument. name need not have to be passed.

**Review by sravantit25 — CHANGES_REQUESTED** [review]

**Comment by sravantit25** [comment]
> This script will fetch the card details for the timeframe ( start date and end date ) with respect to the user as member in the card . Gold card will have the board id , name of the card, Last updated date and board name will be saved to gold tables .
> 
> Added file: drishya_gold_fetch_data.py
> 
> How do we test it: Run from branch python delta-lake-trello/jobs/drishya_fetch_gold_cards.py --start_date 2022-01-01 --end_date 2023-02-24 --name Drishya
> 
> Selection_095

@drishyatm Please update the script to take your name as default. I had mentioned in the instructions to hardcode the name in the script. Each person will have their own gold table and their own script to fetch the data. The idea is to have everyone get accustomed to using Delta tables.

File: jobs/drishya_gold_fetch_data.py (hunk 5 above)
Comment (drishyatm) [reply jobs/drishya_gold_fetch_data.py]: Hardcoded the name now

File: jobs/drishya_gold_fetch_data.py (hunk 1 above)
Comment (drishyatm) [reply jobs/drishya_gold_fetch_data.py]: done!

File: jobs/drishya_gold_fetch_data.py (hunk 2 above)
Comment (drishyatm) [reply jobs/drishya_gold_fetch_data.py]: yes, hardcoded while passing

File: jobs/drishya_gold_fetch_data.py (hunk 3 above)
Comment (drishyatm) [reply jobs/drishya_gold_fetch_data.py]: done!

File: jobs/drishya_gold_fetch_data.py (hunk 4 above)
Comment (drishyatm) [reply jobs/drishya_gold_fetch_data.py]: done!

**Comment by drishyatm** [comment]
Renamed file:
drishya_cards_gold.py

How do we test it:
Run from branch
python delta-lake-trello/jobs/drishya_cards_gold.py --start_date 2022-01-01 --end_date 2023-02-24

Made all the changes as per the suggestions/comments!

File: jobs/drishya_cards_gold.py [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+    :param start_date: trello fetching details start date
+    :param end_date: trello fetching details end date
+    :param nameofuser: the name of user that needs to filter/fetch
+    :return none
+    """
+    path = dc.drishya_cards_gold_path
+    try:
+        # Fetch the cards data from the refined cards silver delta table
+        refined_cards_data = spark.read.format(
+            "delta").load(dc.cards_silver_table_path)
```
Comment (sravantit25) [inline jobs/drishya_cards_gold.py]: The source should be cards_unique_table. Can you change the path to cards_unique_silver_table_path.

File: jobs/drishya_cards_gold.py [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+    Run the steps to get all the data of cards that 
+    are with name drishya and place it Delta Lake Gold table
+    """
+    logger.info(f'Starting the job to fetch {name_user} activity cards')
+    try:
+        # Get SparkSession and SparkContext
+        spark, context = sh.start_spark()
+
+        logger.info(
+            f'Fetched sprint board info for {start_date} between {end_date}')
```
Comment (sravantit25) [inline jobs/drishya_cards_gold.py]: Can you remove the words sprint board here?

**Review by sravantit25 — CHANGES_REQUESTED** [review]

**Comment by sravantit25** [comment]
> Renamed file: drishya_cards_gold.py
> 
> How do we test it: Run from branch python delta-lake-trello/jobs/drishya_cards_gold.py --start_date 2022-01-01 --end_date 2023-02-24
> 
> Made all the changes as per the suggestions/comments!

@drishyatm Thanks Drishya. Looks good. Merging the PR.

**Pull request merged**